tensorflow-core/tensorflow-core-generator/pom.xml

```diff
@@ -28,15 +28,69 @@
       <artifactId>javapoet</artifactId>
       <version>1.12.1</version>
     </dependency>
+    <dependency>
+      <groupId>com.google.protobuf</groupId>
+      <artifactId>protobuf-java</artifactId>
+      <version>${protobuf.version}</version>
+    </dependency>
     <dependency>
       <groupId>com.github.javaparser</groupId>
       <artifactId>javaparser-core</artifactId>
       <version>3.15.12</version>
     </dependency>
+    <!-- https://mvnrepository.com/artifact/org.commonmark/commonmark -->
+    <dependency>
+      <groupId>org.commonmark</groupId>
+      <artifactId>commonmark</artifactId>
+      <version>0.17.1</version>
+    </dependency>
   </dependencies>
 
   <build>
     <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <version>3.0.0</version>
+        <executions>
+          <!--
+          Add generated classes to the source path
+            These are classes generated by the javacpp-plugin, including TensorFlow operations
+            wrappers and C API Java bindings
+          -->
+          <execution>
+            <id>add-gen-sources</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>add-source</goal>
+            </goals>
+            <configuration>
+              <sources>
+                <source>${project.basedir}/src/gen/protoc</source>
+              </sources>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+        <version>3.0.0</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>java</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <mainClass>org.tensorflow.op.generator.OpGenerator</mainClass>
+          <arguments>
+            <argument>${project.basedir}/../tensorflow-core-api/src/gen/java</argument>
+            <argument>${project.basedir}/../tensorflow-core-api/src/gen/resources/org/tensorflow/op/ops.pb</argument>
```

Comment:
Shouldn't the execution of the generator be added to the core's api POM.xml instead? We can build the generator, and we execute it with this maven plugin, passing it the list of arguments.

Reply:
This is as much for manual running/debugging/testing as for the CI.  For the CI, since this is done as part of `build.sh`, I figured I'd just do a `cd ../tensorflow-core-generator && mvn exec`, adding it as part of the POM is complicated (if it's even possible) and unnecessary.

For your top level comment, you can just run the `exec` goal to generate the new ops once you've checked out this branch.  I can make a temp PR with that if you want.

Reply:
The reason why I'd prefer having the exec goal in the core-api pom is that we won't cross anymore module boundaries. You don't need to run this goal from `build.sh`, it can be simply configured to be automatically executed run at the right phase when building the core-api (probably just before we run the annotation processor).

Reply:
Ok, this is done, I left the phase commented out for now so it doesn't auto-run, but it is easy to enable.  We should do that in another PR after this one, with CI Build for testing.